proposed/http-factory/http-factory.md

```diff
@@ -77,24 +82,30 @@ Has the ability to create server requests.
 ```php
 namespace Psr\Http\Message;
 
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UriInterface;
+
 interface ServerRequestFactoryInterface
 {
     /**
      * Create a new server request.
      *
-     * @param string $method
-     * @param UriInterface|string $uri
+     * The method and URI of the request SHOULD be derived from the given
+     * server parameters.
      *
-     * @return ServerRequestInterface
-     */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from PHP globals.
+     * If the method or URI parameter is passed, these values MUST be used
+     * instead of the server parameters.
+     *
+     * @param array $server
+     * @param string|null $method
+     * @param UriInterface|string|null $uri
+     *
+     * @throws \InvalidArgumentException
```

Comment:
Not necessarily; it's varied from one specification to the next. PSR-7 did not define _any_ exceptions, and only specified the most general SPL exceptions that were contextually correct.

That allows implementations to have their own exceptions, so long as they implement the expected exception type from the specification.